apps/server/lib/lexical/server/code_intelligence/entity.ex

```diff
@@ -0,0 +1,240 @@
+defmodule Lexical.Server.CodeIntelligence.Entity do
+  alias Future.Code, as: Code
+  alias Lexical.Document
+  alias Lexical.Document.Location
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.Project
+  alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.CodeIntelligence
+  alias Lexical.RemoteControl.CodeMod
+  alias Lexical.Text
+  alias Sourceror.Zipper
+
+  require Logger
+
+  @type resolved :: {:module, module()}
+
+  @doc """
+  Attempts to resolve the entity at the given position in the document.
+
+  ## Return values
+
+  Returns `{:ok, resolved}` if successful and `{:error, error}` otherwise.
+  Resolved entities are one of:
+
+    * `{:module, module}`
+
+  """
+  @spec resolve(Document.t(), Position.t()) :: {:ok, resolved} | {:error, term()}
+  def resolve(%Document{} = document, %Position{} = position) do
+    with {:ok, ast} <- CodeMod.Ast.from(document),
+         {:ok, zipper} <- innermost_zipper_at(ast, position),
+         {:ok, resolved} <- resolve(Zipper.node(zipper), zipper, document, position) do
+      Logger.info("Resolved entity: #{inspect(resolved)}")
+      {:ok, resolved}
+    else
+      error ->
+        Logger.info("Resolve failed: #{inspect(error)}")
+        error
+    end
+  end
+
+  defp resolve({:__block__, _, [atom]} = node, zipper, document, position) when is_atom(atom) do
+    parent = Zipper.up(zipper)
+
+    case Zipper.node(parent) do
+      {:__aliases__, _, _} = aliases ->
+        resolve(aliases, parent, document, position)
```

Comment:
There are three functions in code that I think can help you here, `Code.Fragment.cursor_context`, `Code.Fragment.container_context` and `Code.Fragment.container_context_to_quoted`. 

I think most of this can be accomplished by trimming the string to the position and giving the result to `Code.Fragment.cursor_context` or by using `Code.Fragment.container_context`. Apologies if you've already tried these functions and rejected them, but I'd like to know why.

I played around a bit with `Code.Fragment.container_context` and it was particularly good at resolving modules, popping out an alias which you can use the alias resolver to expand.